driver.go

```diff
@@ -98,6 +109,28 @@ func (driver *linodeVolumeDriver) determineLinodeID() error {
 		return nil
 	}
 
+	return driver.determineLinodeIDFromLabel()
+}
+
+func (driver *linodeVolumeDriver) determineLinodeIDFromMetadata() error {
+	client, err := metadata.NewClient(context.Background())
```

Comment:
I noticed it takes a bit for an error to be raised here when running on a non-MDS region. Maybe we could manually dial the MDS API with a short timeout  and check for that error in `determineLinodeID`?

e.g.
```go
conn, err := net.DialTimeout("tcp", fmt.Sprintf("%s:80", metadata.APIHost), 2*time.Second)
if err != nil {
	return err
}

conn.Close()
```